File: ios/components/image/imageView/YDImageViewManager.m
//
//  YDImageViewManager.m
//
//  Created by jackie on 2020/8/20.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"


@interface RCT_EXTERN_MODULE(YDImageViewManager, RCTViewManager)


// 指明显示的图片名称，images.xcassets目录中
RCT_EXPORT_VIEW_PROPERTY(fileName, NSString)

// 用于显示网络图片
RCT_EXPORT_VIEW_PROPERTY(filePath, NSString)
RCT_EXPORT_VIEW_PROPERTY(defaultImageName, NSString)

// 图片的圆角
RCT_EXPORT_VIEW_PROPERTY(cornerRadius, NSNumber)

// 图片的点击预览属性
RCT_EXPORT_VIEW_PROPERTY(previewable, NSNumber)



// 更新网络图片，兼容性处理
RCT_EXTERN_METHOD(onUpdate:(nonnull NSNumber *)node content:(nonnull NSString *)content)

@end

File: ios/components/image/imageView/YDImageViewManager.swift
//
//  YDImageViewManager.swift
//
//  Created by jackie on 2020/8/20.
//

import Foundation


@objc(YDImageViewManager)
class YDImageViewManager: RCTViewManager {
  
  
  override func view() -> UIView! {
    let imageView = YDImageView(frame: CGRect.zero)
    imageView.contentMode = .scaleAspectFill
    imageView.clipsToBounds = true
    return imageView
  }
  
  @objc public func onUpdate(_ node: NSNumber, content: NSString) {
    // 过滤掉无效的content
    let contentStr = content as String
    if (contentStr.count <= 0 || !contentStr.contains("http")) {
      return
    }
    // 主线程上更新imageview控件的网络图片路径信息
    DispatchQueue.main.async {
      if let imageView = self.bridge.uiManager.view(forReactTag: node) as? YDImageView {
        imageView.filePath = content
        imageView.onUpdate()
      }
    }
  }
  
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  
  
  private func getTargetView(node: NSNumber) -> YDImageView? {
    if let view = self.bridge.uiManager.view(forReactTag: node) {
      return view as? YDImageView
    }
    return nil
  }
  
  
}
